docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
 
-{Describe the value proposition: what problem does it solve?}
+--------------------------------------------------------------------------------------------------------------------
 
-## User Stories
+## 2. Setting Up
+
+### 2.1. Prerequisites
+* JDK 11
+* IntelliJ IDEA
+
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` → `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle.
+    1. Click `Configure` → `Project Defaults` → `Project Structure`.
+    2. Click `New…` and set it to the directory of the JDK.
+4. Click `Import Project` (or `Open or Import` in newer version of Intellij).
+5. Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+If asked, choose to `Open as Project` (not `Open as File`).
+7. Click `OK` to accept the default settings
+
+--------------------------------------------------------------------------------------------------------------------
+
+## 3. Design (Architecture)
+(Zeyu)
+
+![Architecture Diagram of Design Component](UML/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+The main class is `Kaji.java`. It is responsible for:
+
+* Initializing the components in the correct sequence, and connects them up with each other at app launch.
+* Shutting down the components and invoking cleanup methods where necessary at exit.
+
+The rest of the App consists of 8 components:
+* `Access`: Keeps track of the access level of the user.
+* `UI`: The UI of the App.
+* `Command`: Executes the different command types.
+* `Common`: Contains common classes.
+* `Manager`: Holds the data of the App in memory.
+* `Paser`: Parses user input into specific command type.
+* `Scheduler`: Schedules the revision schedule.
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+### 3.1. Ui Component 
+(Jia Ern)
+
+![Class Diagram of Ui Component](DG_Images/ui_component.PNG)
+
+The UI component consists of a main class — Ui.java. The Kaji, Logic and Storage components have a dependency on the UI component due to the need to take in user input and show the results of execution.
+
+The UI component is responsible for:
+* Taking in user input.
+* Printing result of execution for the different commands. 
+* Printing error messages when certain exceptions occur. 
+
+### 3.2. Logic Component 
+(Jane)
+
+The Logic component consists of the `Parser`, `Command` and `Scheduler` classes.
+
+<p align="center">
+  <img src="DG_Images/LogicClassDiagram.png" width="600" alt="Logic Class Diagram"/>
```

Comment:
Maybe in your diagram, you can consider having labels for the other lines since you have already had them for some of the lines.
![Screenshot_2](https://user-images.githubusercontent.com/54029207/97529481-f1850380-19ea-11eb-804d-f3f7ec589f30.png)